src/new/wallet.js

```diff
@@ -765,44 +765,128 @@ class HathorWallet extends EventEmitter {
+    // then the first step is to separate the inputs/outputs for each token
+    const defaultData = { outputs: [], inputs: [] };
+    if (txToken && txToken.uid !== HATHOR_TOKEN_CONFIG.uid) {
+      tokensData[txToken.uid] = Object.assign({}, defaultData);
+      tokens.push(txToken.uid);
+    } else {
+      for (const output of outputs) {
+        if (output.token && output.token !== HATHOR_TOKEN_CONFIG.uid) {
+          if (!(output.token in tokensData)) {
+            tokens.push(output.token);
+            tokensData[output.token] = Object.assign({}, defaultData);
+          }
+        }
+      }
+
+      if (tokens.length === 0) {
+        // So it's HTR
+        tokensData[HATHOR_TOKEN_CONFIG.uid] = Object.assign({}, defaultData);
+      }
+    }
 
     for (const output of outputs) {
-      data.outputs.push({address: output.address, value: output.value, tokenData: isHathorToken ? 0 : 1})
+      let tokenData;
+      if (txToken && txToken.uid !== HATHOR_TOKEN_CONFIG.uid) {
+        // A single token with third method parameter of class default
+        tokenData = 1;
+      } else {
+        if (tokens.length === 0) {
+          // HTR
+          tokenData = 0;
+        } else {
+          tokenData = tokens.indexOf(output.token) + 1;
+        }
+      }
+
+      tokensData[output.token].outputs = [...tokensData[output.token].outputs, {
+        address: output.address,
+        value: output.value,
+        timelock: output.timelock ? output.timelock : null,
+        tokenData,
+      }];
     }
 
     for (const input of inputs) {
-      data.inputs.push({tx_id: input.hash, index: input.index, token: HATHOR_TOKEN_CONFIG.uid });
+      let token;
+      if (txToken && txToken.uid !== HATHOR_TOKEN_CONFIG.uid) {
+        // A single token with third method parameter of class default
+        token = txToken.uid;
+      } else {
+        if (tokens.length === 0) {
+          // HTR
+          token = HATHOR_TOKEN_CONFIG.uid;
+        } else {
+          token = input.token;
+        }
+      }
+
+      tokensData[input.token].inputs = [...tokensData[input.token].inputs, {
+        tx_id: input.hash,
+        index: input.index,
+        token,
+      }];
     }
 
-    const ret = this.completeTxData(data, txToken, options);
+    const fullTxData = Object.assign({tokens}, defaultData);
 
-    if (ret.success) {
-      return this.sendPreparedTransaction(ret.data);
-    } else {
-      return ret;
+    const walletData = wallet.getWalletData();
+    const historyTxs = 'historyTransactions' in walletData ? walletData.historyTransactions : {};
+    for (const tokenUid in tokensData) {
+      // For each token key in tokensData we prepare the data
+      const partialData = tokensData[tokenUid];
+      let chooseInputs = true;
+      if (partialData.inputs.length > 0) {
+        chooseInputs = false;
+      }
+
+      // Warning: prepareSendTokensData(...) might modify `partialData`.
```

Comment:
What changes might occur?

Reply:
This method may add inputs in the inputs array, if `chooseInputs=true` and also add the change output.

Reply:
Done